src/EntryPoints/Specials/SpecialManageApprovers.php

```diff
@@ -109,10 +107,19 @@ private function approversToViewModel( array $approvers ): array {
 	 * @return array<string, mixed>
 	 */
 	private function approverToViewModel( Approver $approver ): array {
+		$showCategories = true;
+
+		if ( empty( $approver->categories ) ) {
+			$request = $this->getRequest();
+			$user = $this->userFactory->newFromName( $request->getText( 'username' ) );
+			$showCategories = $request->wasPosted() && $user->getName() === $approver->username;
+		}
+
 		return [
 			'username' => $approver->username,
 			'userId' => $approver->userId,
-			'categories' => $approver->categories
+			'categories' => $approver->categories,
+			'showCategories' => $showCategories
```

Comment:
This is poorly named. It's not indicating if categories should be shown

Reply:
Though why output this data structure at all, to then not use it? I think it makes more sense to filter out the Approver without categories from the Approver[]. No changes needed on the frontend. Also easier to test